### ISSUE TITLE
Proxy detection mechanism only recognises uppercase environment variables

### ISSUE THREAD

**Issue by ghost** [wildfoundry/dataplicity-lomond#78]
Hello!

Current implementation as in `WebSocket._detect_proxies`  class method will only detect uppercase proxy server environment variables. This is sub-optimal, since majority of software respects lowercase `http_proxy` and `https_proxy` environment variables.

Luckily case insensitive functionality is already available in Python's standard library, specifically in `urllib.request.getproxies` https://docs.python.org/3/library/urllib.request.html?highlight=getproxies#urllib.request.getproxies.

Please, consider using `urllib.request.getproxies()` for proxy detection instead of custom solution!
Thanks!

**Comment by willmcgugan** [comment]
Thanks, I wasn't aware of `geproxies` in urllib. Looks like I'll need to implement a Python2.7 version, as it is Py3 only.

**Comment by ghost** [comment]
@willmcgugan Right, did not think about 2.7 compatibility. Well, Python 2.7 has `urllib.proxies()` https://docs.python.org/2.7/library/urllib.html?highlight=getproxies#urllib.getproxies

And I bet `six` can take care of that.

**Comment by barfle** [comment]
Thanks for this report. I have implemented a fix to use environment proxy discovery that honors lowercase and uppercase env vars ( / , etc.) and added regression tests. Closing as resolved.
